### ISSUE TITLE
Clean Material Slots broken in Blender 3

### ISSUE THREAD

**Issue by Vinc3r** [ChrisHinde/MaterialUtilities#10]
**Describe the bug**
When an object use more than 2 materials, and have in addition to them some materials unused, Clean Material Slots is messy and unassign some materials

**To Reproduce**
1. assign 2 materials on different parts of a mesh
2. create a 3rd material, without assigning it in any face
3. call the Clean Material Slots function
4. materials are now messy

[cleanmtl.blend.zip](https://github.com/ChrisHinde/MaterialUtilities/files/8031800/cleanmtl.blend.zip)

**Expected behavior**
3rd material & slot should be erased

**Screenshots**

![cleanmtl](https://user-images.githubusercontent.com/3873421/153181015-dfb5972c-8587-479b-b9c9-87d5a2c454d8.gif)

**Versions:**
 - Blender 3.0.1 (official release), branch: master, commit date: 2022-01-25 17:19 hash:dc2d18018171
 - Add-on Version 2.4.0

**Comment by simonwidjaja** [comment]
This plugin does not work in blender 3.x anymore. Always getting the same error:

Python: Traceback (most recent call last):
  File "/Applications/Blender.app/Contents/Resources/3.0/scripts/addons/materials_utils/menus.py", line 56, in draw
    icon_value = material.preview.icon_id)
AttributeError: 'NoneType' object has no attribute 'icon_id'

Are there any plans to fix this? Should be easily fixed. 

Here is my interim script solution that works like a charme:

```
import bpy
    
for ob in bpy.context.selected_editable_objects:
    ob.active_material_index = 0
    for i in range(len(ob.material_slots)):
        bpy.ops.object.material_slot_remove({'object': ob})
```

**Comment by ChrisHinde** [comment]
The Clean Material Slots is now fixed (Commit [887ed70](https://github.com/ChrisHinde/MaterialUtilities/commit/887ed7012deed6219c7136e29edf6d93b8b0dafd) )

The problem that created the error message @simonwidjaja  posted is also fixed (Commit [db9522f](https://github.com/ChrisHinde/MaterialUtilities/commit/db9522feafa9689162ea6f3453dbea25781d2336) )

**Comment by Vinc3r** [comment]
as simply as that! thanks for the fix
